api/api_errors.py

```diff
@@ -0,0 +1,35 @@
+import django
+django.setup()
+from sefaria.model import *
+from typing import List
+
+
+class APIError():
```

Comment:
The way this has been implemented is great. Very orderly and clear. 
That being said, I have a concern we might be giving the impression that this is used similarly to an exception, which it is not. The naming scheme here may cause some confusion. Suggest we change all errors to "response"? 

Also The subclasses all seem to be unique to the texts api, so I would suggest we further group them under a parent like that. Otherwise in the future this might become more and more confusing for incoming developers to use.

Reply:
you're right, but we're using them to flag some 'error', and Response does not reflect it. also response is a 'saved' term for the whole http response.
i'm thinking aloud - we're using it for flagging that some (or all) of the request includes things we're missing. so maybe APIMissingMessage or APILackMessage? @edamboritz @akiva10b what do you think?

Reply:
we spoke about this earlier as well, is a bad ref or no version really an "error", it it an HTTP 404 error or just an empty response?
This is an area with tradeoffs and I would vote for whatever jives best with what devs are familiar with from API documentation. For instance, in https://www.weatherapi.com/docs/ which is somewhat similar to us in the sense that you can ask for data for multiple parts and only receive some, "errors" only refer to something malfunctioning in the API or the constraints around it (expires key, bad json formatting etc.)
The unifying metric here is "unavailableData" so I would name the class UnavailableDataResponse and so on for the rest of the classes NoVersionResponse... @YishaiGlasner what do you think?

Reply:
it likely also makes sense to put that data in a "unavailable_data" parameter in the API (we are putting it in "errors" now I think which is not 100% correct)

Reply:
i think respone is not the right word here. it's a part of the response. UnavailabeDataMessage?